reflection/Reflection/Reflection/classes/ConfigurationComponentBase.cs

```diff
@@ -0,0 +1,92 @@
+using System;
+using System.Configuration;
+using System.Reflection;
+using Reflection.attributes;
+
+namespace Reflection.classes
+{
+    public class ConfigurationComponentBase
+    {
+        public void SaveSettings()
+        {
+            Type type = GetType();
+            PropertyInfo[] properties = type.GetProperties();
+
+            foreach (PropertyInfo property in properties)
+            {
+                ConfigurationItemAttribute attribute = property.GetCustomAttribute<ConfigurationItemAttribute>();
+                if (attribute == null)
+                    continue;
+
+                object value = property.GetValue(this);
+
+                if (attribute.ProviderType != null && typeof(IConfigurationProvider).IsAssignableFrom(attribute.ProviderType))
+                {
+                    try
```

Comment:
rework this try catch block similar way you've implemented in LoadSettings() method

Reply:
what would even better, if you move logic of creating provider to a separate private method and reuse it in Load and Save methods